packages/graphiql/src/components/GraphiQL.js

```diff
@@ -39,6 +39,19 @@ import {
 
 const DEFAULT_DOC_EXPLORER_WIDTH = 350;
 
+// eslint-disable-next-line no-console
+const logger = console.log;
+
+if (!React.version || !React.version.indexOf('16') < 0) {
+  logger.warn(
+    [
+      'GraphiQL 0.18.0 and after is not compatible with React 15 or below.',
+      'If you are using a CDN source (jsdelivr, unpkg, etc), follow this example:',
+      'https://github.com/graphql/graphiql/blob/master/examples/graphiql-cdn/index.html#L49',
+    ].join('\n'),
```

Comment:
This error is misleading; e.g. if they were to use React 17 or 18 in the future.

Reply:
we will likely remove this before react 17. i can add a todo for that